platform/android/display_server_android.cpp

```diff
@@ -659,17 +664,75 @@ void DisplayServerAndroid::process_hover(int p_type, Point2 p_pos) {
 	}
 }
 
-void DisplayServerAndroid::process_double_tap(Point2 p_pos) {
+void DisplayServerAndroid::process_mouse_event(int p_action, int p_button_mask, Point2 p_pos, float factor) {
+	switch (p_action) {
+		case AMOTION_EVENT_ACTION_BUTTON_PRESS:
+		case AMOTION_EVENT_ACTION_BUTTON_RELEASE: {
+			Ref<InputEventMouseButton> ev;
+			ev.instance();
+			ev->set_position(p_pos);
+			ev->set_global_position(p_pos);
+			ev->set_pressed(p_action == AMOTION_EVENT_ACTION_BUTTON_PRESS);
+			ev->set_factor(factor);
+			int button_mask = buttons_state ^ p_button_mask;
```

Comment:
What is `buttons_state ^ p_button_mask` for?

Reply:
Get currently changed button, as xor gives difference between bitmasks

Reply:
Makes sense! Maybe renaming `button_mask` to `changed_button_mask` could make it perfectly clear when reading the code.

Reply:
Well, it's event button_mask so it made sense in my head 😅 but I'll change that if that makes it more readable